index.js

```diff
@@ -38,25 +40,37 @@ async function start (opts) {
     stop
   }
 
-  res.app.server.on('request', res.app.server.handler)
+  res.app.server.on('request', handler)
 
   return res
 
   async function restart (_opts = opts) {
     const old = res.app
-    const oldHandler = serverWrapper.server.handler
+    const oldHandler = handler
     const clientErrorListeners = old.server.listeners('clientError')
     const newApp = spinUpFastify(_opts, serverWrapper, restart)
-    await newApp.ready()
-    old.server.removeListener('request', oldHandler)
-    old.server.removeListener('request', oldHandler)
-    newApp.server.on('request', newApp.server.handler)
-    for (const listener of clientErrorListeners) {
-      old.server.removeListener('clientError', listener)
-    }
-    res.app = newApp
+    try {
+      await newApp.ready()
+      old.server.removeListener('request', oldHandler)
+      old.server.removeListener('request', oldHandler)
+      newApp.server.on('request', handler)
+      for (const listener of clientErrorListeners) {
+        old.server.removeListener('clientError', listener)
+      }
+      res.app = newApp
 
-    await old.close()
+      await old.close()
+    } catch (err) {
+      const listenersNow = newApp.server.listeners('clientError')
```

Comment:
I don't understand why we need to update the listener's list. Why can't we use `clientErrorListeners`?

Reply:
I've added a comment explaining this.